js_modules/dagster-ui/packages/ui-core/src/runs/useCursorAccumulatedQuery.tsx

```diff
@@ -91,25 +91,27 @@ export function useCursorAccumulatedQuery<
 }: {
   query: DocumentNode;
   variables: Omit<TVars, 'cursor'>;
-  getResult: (responseData: TQuery) => FetchResult<DataType, CursorType, ErrorType>;
+  // Important: getResult must be memoized!
+  getResult: (responseData: TQuery) => AccumulatingFetchResult<DataType, CursorType, ErrorType>;
 }) {
   const [fetched, setFetched] = useState<DataType[] | null>(null);
   const [error, setError] = useState<ErrorType | null>(null);
   const client = useApolloClient();
 
+  const variablesJSON = JSON.stringify(variables || {});
   const {stop, fetch} = useMemo(() => {
     return new AccumulatingDataFetcher({
       fetchData: async (cursor) => {
         const resp = await client.query<TQuery, TVars>({
-          variables: {...variables, cursor} as TVars,
+          variables: {...JSON.parse(variablesJSON), cursor} as TVars,
```

Comment:
I don't think this is relevant to the PR anymore, but it was a tricky bug to find -- Apollo useQuery behaves this way and I think it's best that this match